lib/MiniDroneBtAdapter.js

```diff
@@ -305,16 +400,26 @@ class MiniDroneBtAdapter extends EventEmitter {
         if (!peripheral) {
             return false;
         }
-
         const localName = peripheral.advertisement.localName;
         const manufacturer = peripheral.advertisement.manufacturerData;
-        const localNameMatch = localName && DRONE_PREFIXES.some((prefix) =>
-            localName.indexOf(prefix) >= 0
-        );
-        const manufacturerMatch = manufacturer && (MANUFACTURER_SERIALS.indexOf(manufacturer) >= 0);
 
-        // Is true for EITHER an "RS_" name OR manufacturer code.
-        return localNameMatch || manufacturerMatch;
+        if (this.options.droneFilter == '') {
+          const localNameMatch = localName && DRONE_PREFIXES.some((prefix) =>
+              localName.indexOf(prefix) >= 0
+          );
+          const manufacturerMatch = manufacturer && (MANUFACTURER_SERIALS.indexOf(manufacturer) >= 0);
+
+          // Is true for EITHER an "RS_" name OR manufacturer code.
+          return localNameMatch || manufacturerMatch;
+        }
+        else {
+          if (localName == this.options.droneFilter) {
+            return true;
+          }
+          else {
+            return false;
+          }
+        }
```

Comment:
This code didn't work as expected, it is true for a bad DroneFilter as soon as a good prefix is present in the DroneFilter.
I simplified the working code this way, I hope it is simpler :)
https://github.com/gotleg/npm-parrot-minidrone/commit/5eb88600ee235b70a1a4fb8aa6388e1f3f08ef1d